noir_stdlib/src/hash.nr

```diff
@@ -1,36 +1,37 @@
+
 #[foreign(sha256)]
-fn sha256(_input : [u8]) -> [u8; 32] {}
+fn sha256<N>(_input : [u8; N]) -> [u8; 32] {}
 
 #[foreign(blake2s)]
-fn blake2s(_input : [u8]) -> [u8; 32] {}
+fn blake2s<N>(_input : [u8; N]) -> [u8; 32] {}
```

Comment:
Maybe I'm missing something, why was this change made here?

Reply:
change referring to adding `N`

Reply:
I was changing the arrays with anonymous/inferred lengths to ones with explicit lengths in the code that I was touching. It is done to encourage users themselves to use explicit lengths since the inferred ones are still buggy (ref #562). Since users may look into the stdlib to inspect the code or figure out best practices I figured while I was already touching the code I might as well change it.

Reply:
Yeah that makes sense and thanks for fixing it for the other functions